python/ray/data/datasource/parquet_datasource.py

```diff
@@ -444,15 +444,19 @@ def _sample_piece(
     # Use first batch in-memory size as ratio estimation.
     try:
         batch = next(batches)
-        in_memory_size = batch.nbytes / batch.num_rows
-        metadata = piece.metadata
-        total_size = 0
-        for idx in range(metadata.num_row_groups):
-            total_size += metadata.row_group(idx).total_byte_size
-        file_size = total_size / metadata.num_rows
-        ratio = in_memory_size / file_size
     except StopIteration:
         ratio = PARQUET_ENCODING_RATIO_ESTIMATE_LOWER_BOUND
+    else:
```

Comment:
Is this affected somehow or just a drive-by polish?

Reply:
Yes, we weren't handling the sampling case where the returned batch was empty, which was causing a failure here; this was a driveby fix.